src/api/index.js

```diff
@@ -1,44 +1,42 @@
-import fetch from 'isomorphic-fetch'
 import config from './config'
+import fetch from 'isomorphic-fetch'
 
 function callApi(url, method = 'get') {
   return fetch(`${url}`, {
     headers: config.headers,
     method
   })
-  .then(response => response.json().then(json => ({ json, response })))
-  .then(({ json, response }) => {
-    if (!response.ok) {
-      return Promise.reject(json);
-    }
+  .then((response) => {
+      if(!response.ok) {
+        return Promise.reject(response)
+      }
 
-    return json;
+      return response.json()
   })
-  .then(
-    response => response,
-    error => error
-  );
+  .catch((response) => console.log(`API call failed because: ${response.status} ${response.statusText}`))
 }
 
 // Returns a user by their username
 export function getSummonerByName(name) {
-	const URL = `https://na.api.pvp.net/api/lol/na/v1.4/summoner/by-name/${name}?api_key=${config.key}`
-	return callApi(URL)
+  const URL = `https://na.api.pvp.net/api/lol/na/v1.4/summoner/by-name/${name}?api_key=${config.key}`
+  return callApi(URL).then((response) => response[name])
 }
 
 // Returns the match list for a given user id. Not doing it now but we'd also filter based on start + end
-export function getMatchlist(id) {
-	// obviously can enter query params (start + end in 'epoch milliseconds')
-	const URL = `https://na.api.pvp.net/api/lol/na/v2.2/matchlist/by-summoner/${id}?api_key=${config.key}`
-	return callApi(URL)
+export function getMatchList(summoner) {
```

Comment:
Changing these functions to accept summoners instead of id's allows to you to do stuff like:

```
getSummonerByName('roflcopter').then(getMatchList)
```